presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -107,7 +120,59 @@ public void process(Iterable<QueryExecution> queries)
         return ImmutableMap.copyOf(pools);
     }
 
-    private void updateNodes()
+    private MemoryPoolAssignmentsRequest updateAssignments(Iterable<QueryExecution> queries)
+    {
+        ClusterMemoryPool reservedPool = pools.get(RESERVED_POOL);
+        long version = memoryPoolAssignmentsVersion.incrementAndGet();
+        if (reservedPool != null && allAssignmentsHavePropagated(queries)) {
+            if (reservedPool.getQueries() == 0) {
+                QueryExecution biggestQuery = null;
```

Comment:
you could use streams for this:
```java
        QueryExecution biggestQuery = queries.stream()
                .max((left, right) -> left.getQueryInfo()
                        .getQueryStats()
                        .getTotalMemoryReservation()
                        .compareTo(right.getQueryInfo().getQueryStats().getTotalMemoryReservation()))
                .get();
        long maxMemory = biggestQuery.getQueryInfo().getQueryStats().getTotalMemoryReservation().toBytes();
```
But it would call `getQueryInfo` a lot of times which is not cheap